src/approaches/nsrt_rl_approach.py

```diff
@@ -49,33 +65,215 @@ def get_interaction_requests(self) -> List[InteractionRequest]:
             task = self._train_tasks[i]
             try:
                 _act_policy = self.solve(task, CFG.timeout)
+                self._train_task_to_option_plan[i] = self._last_plan
             except (ApproachTimeout, ApproachFailure) as e:
                 partial_refinements = e.info.get("partial_refinements")
                 assert partial_refinements is not None
                 _, plan = max(partial_refinements, key=lambda x: len(x[1]))
                 _act_policy = utils.option_plan_to_policy(plan)
+                self._train_task_to_option_plan[i] = plan
             request = InteractionRequest(train_task_idx=i,
                                          act_policy=_act_policy,
                                          query_policy=lambda s: None,
                                          termination_function=task.goal_holds)
             requests.append(request)
         return requests
 
+    @staticmethod
+    def _get_unchanging_features(
+            changing_var_to_feat: Dict[Variable, List[int]],
+            var_to_obj: VarToObjSub) -> Dict[Variable, List[int]]:
+        var_to_unchanging_feat_ind = {}
+        for var, changing_indices in changing_var_to_feat.items():
+            dim = var_to_obj[var].type.dim
+            unchanging_indices = [
+                i for i in range(dim) if i not in changing_indices
+            ]
+            var_to_unchanging_feat_ind[var] = unchanging_indices
+        return var_to_unchanging_feat_ind
+
     def learn_from_interaction_results(
```

Comment:
this has become a very long method, can we split it up into some helper methods for easier readibility?